contracts/tokenomics/maker/src/utils.rs

```diff
@@ -130,3 +138,56 @@ pub fn validate_bridge(
 
     Ok(bridge_pool)
 }
+
+/// # Description
+/// Checks if from_token - UST pool exists
+/// # Params
+/// * **deps** is the object of type [`DepsMut`].
+///
+/// * **cfg** is the object of type [`Config`].
+///
+/// * **from_token** is the object of type [`AssetInfo`].
+pub fn uusd_pool_exists(deps: Deps, cfg: &Config, from_token: AssetInfo) -> Option<PairInfo> {
+    let uusd = native_asset_info(UUSD_DENOM.to_string());
+
+    let uusd_pool = query_pair_info(
+        &deps.querier,
+        cfg.factory_contract.clone(),
+        &[from_token.clone(), uusd.clone()],
+    );
+
+    if !uusd_pool.is_ok() {
+        return None;
+    }
+
+    return Some(uusd_pool.unwrap());
+}
+
+/// # Description
+/// /// Checks if from_token - luna - ust pool exists
+/// # Params
+/// * **deps** is the object of type [`DepsMut`].
+///
+/// * **cfg** is the object of type [`Config`].
+///
+/// * **from_token** is the object of type [`AssetInfo`].
+pub fn uluna_uusd_pools_exist(deps: Deps, cfg: &Config, from_token: AssetInfo) -> Option<PairInfo> {
```

Comment:
I think this function is a clone of `uusd_pool_exists()`. Lets combine them into one function with one additional parameter: `native_asset_info`